src/com/epam/javatrainings/annotations/processor/EmailAnnotationProcessor.java

```diff
@@ -0,0 +1,21 @@
+package com.epam.javatrainings.annotations.processor;
+
+import com.epam.javatrainings.annotations.exception.EmailNotValidException;
+
+import java.lang.reflect.Field;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class EmailAnnotationProcessor {
+    public <T> EmailAnnotationProcessor(Field field, T t) throws EmailNotValidException, IllegalAccessException {
+        field.setAccessible(true);
+        String regex = "^[a-zA-Z0-9_!#$%&'*+/=?`{|}~^-]+(?:\\.[a-zA-Z0-9_!#$%&'*+/=?`{|}~^-]+)*@[a-zA-Z0-9-]+(?:\\.[a-zA-Z0-9-]+)*$";
+        Pattern pattern = Pattern.compile(regex);
```

Comment:
Should be declared on class level, because the regex must be compiled once

Reply:
good)